cpp/ql/test/query-tests/Security/CWE/CWE-134/semmle/consts/NonConstantFormat.expected

```diff
@@ -1,16 +1,10 @@
-| consts.cpp:63:9:63:10 | c5 | The format string argument to printf should be constant to prevent security issues and other potential errors. |
-| consts.cpp:69:9:69:10 | c6 | The format string argument to printf should be constant to prevent security issues and other potential errors. |
 | consts.cpp:81:9:81:10 | c8 | The format string argument to printf should be constant to prevent security issues and other potential errors. |
-| consts.cpp:86:9:86:10 | v1 | The format string argument to printf should be constant to prevent security issues and other potential errors. |
```

Comment:
I think the problem with line 86 is that line 84 is wrong. Instead of `char *v1;`, it should say something like `char v1[100];`. It could reasonably also say `char *v1 = malloc(100);`, but the data flow library doesn't support that; it's not considered to be a definition by reference.

There's the same problem on line 89 of `consts.cpp`.

Reply:
I assumed that `consts.cpp` was somehow tied to the CWE suite and hence was not subject to editing.  Am I mistaken here?

As far as line 84 and its subsequent uses, I do not believe it is wrong:  it is meant to test whether we are handling definition by reference correctly.  Presently, it appears that we are not (which is why I left a `TODO` comment next to `DataFlow::DefinitionByReferenceNode`).  What do others think?

Reply:
@geoffw0, @jbj tells me that you typically maintain the CWE test suite.  Is it OK for me to modify files therein?

Reply:
`consts.cpp` is associated with a CWE but the test cases were written by us, and can be edited by us.